cg/src/domains/management/subjects/subjects.controller.js

```diff
@@ -1,12 +1,48 @@
 const SubjectsService = require('./subjects.service');
 
+const PAGINATION_COUNT = 2; // HARD CODED CONSTANT
+
 class SubjectsController {
   constructor(subjectsService = null) {
     this.subjectsService = subjectsService || new SubjectsService();
   }
 
   async listSubjects(req, res) {
-    res.json(await this.subjectsService.listSubjects());
+    const subjectsList = await this.subjectsService.listSubjects();
+    var requestedPage = req.query.page;
+    var pages = [];
+
+    if (requestedPage === undefined) {
+      requestedPage = 1;
+    }
+
+    for (var i = 0; i * PAGINATION_COUNT < subjectsList.length; i++) {
+      pages[i] = subjectsList.slice(i * PAGINATION_COUNT, (i + 1) * PAGINATION_COUNT);
+    }
+
+    if (i == 0) {
+      // This block handles the case in which there are no subjects in the db
+      if (requestedPage == 1) {
```

Comment:
I don't think this logic is needed if you use DB pagination

Reply:
Ok I will look into it